Prog/Eviction Group/Pulse-housing-analysis.R

```diff
@@ -0,0 +1,141 @@
+                              "Pressure", # Respondents can choose multiple reasons
+                              ifelse(MOVEWHY8 == 1, "No pressure", "Not reported")),
+           increase_rent = case_when(MOVEWHY1 == 1 ~ "Increased rent"), 
+           missed_rent = case_when(MOVEWHY2 == 1 ~ "Missed Rent"), 
+           repairs_not_made = case_when(MOVEWHY3 == 1 ~ "Repairs not made"),
+           eviction_threatened = case_when(MOVEWHY4 == 1 ~ "Eviction threatened"),
+           locks_changed = case_when(MOVEWHY5 == 1 ~ "Locks changed"),
+           nhbd_danger = case_when(MOVEWHY6 == 1 ~ "Neighborhood dangerous"),
+           other_pressure = case_when(MOVEWHY7 == 1 ~ "Other pressure"),
+           moved = case_when(MOVED == 1 ~ "Moved from Pressure", 
+                             MOVED == 2 ~ "Did not move from Pressure",
+                             MOVED == -99 | MOVED == -88 ~ "Did not report"),
+           income = case_when(INCOME == 1 ~ "$25,000 below",  
+                            INCOME == 2 | INCOME == 3 ~ "$25,000-$49,999",
+                            INCOME == 4 ~ "$50,000-$74,999",
+                            INCOME == 5 ~ "$75,000-$99,999",
+                            INCOME == 6 | INCOME == 7 | INCOME == 8 ~ "$100,000 above",
+                            INCOME == -99 | INCOME == -88 ~ "Not reported"),
+           rent_behind = case_when(RENTCUR == 1 ~ "Not behind on rent",
+                                   RENTCUR == 2 ~ "Behind on rent",
+                                   RENTCUR == -99 | RENTCUR == -88 ~ "Did not report"),
+           eviction_two_months = case_when(EVICT == 1 ~ "very likely", 
+                                           EVICT == 2 ~ "somewhat likely", 
+                                           EVICT == 3 ~ "not very likely", 
+                                           EVICT == 4 ~ "not likely at all",
+                                           EVICT == -99 | EVICT == -88 ~ "Did not report"),
+           inc_cat=case_when(INCOME == 1 | INCOME == 2 | INCOME == 3 ~ "below 40 AMI", #0-$49,999; #based on HUD 2024 incomes limits for income & household size 
+                             INCOME == 4 & THHLD_NUMPER >= 2 ~ "below 40 AMI", #50,000-75,000 & 2 person household or more
+                             INCOME == 5 & THHLD_NUMPER >= 7 ~ "below 40 AMI", #75-100k & 7 person household or more
+                             INCOME == -99 | INCOME == -88 ~ "Not reported",
+                             TRUE ~ "above 40 AMI")) %>%
+  select(SCRAM,HWEIGHT,PWEIGHT,WEEK,CYCLE,EST_ST,
+         pressured,moved,increase_rent,missed_rent,repairs_not_made,eviction_threatened,
+         locks_changed,nhbd_danger,other_pressure,
+         rent_behind,eviction_two_months,income,inc_cat,THHLD_NUMPER, INCOME) 
+
+all_PUF <- inner_join(clean_data, rep_weights, by = c("SCRAM", "WEEK", "CYCLE")) # need to join by week/cycle for replicate weights
```

Comment:
This inner_join() code only keeps rows that are in both `clean_data` and `rep_weights`. The below code revealed that 2851 `SCRAM` values from `clean_data` (out of 3382) exist in the `rep_weights` dataset. The resulting merged dataset `all_PUF` has 2851 entries.

Reply:
I have the same num of obs for clean_data and all_PUF, which I assumed meant that all the needed weights joined with the cleaned data.